Fetcher panics sometimes in CI

E.g. there are two such failures in https://github.com/sourcenetwork/defradb/actions/runs/12874748785/job/35894769441?pr=3396 `TestPushlog_WithValidPeerID_NoError`

```
=== FAIL: net TestPushlog_WithValidPeerID_NoError (unknown)
Jan 20 19:45:16.708 INF net Created LibP2P host PeerId=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW Address=[/ip4/127.0.0.1/tcp/0]
Jan 20 19:45:16.709 INF net Starting internal broadcaster for pubsub network
Jan 20 19:45:16.741 INF net Created LibP2P host PeerId=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Address=[/ip4/127.0.0.1/tcp/0]
Jan 20 19:45:16.741 INF net Starting internal broadcaster for pubsub network
Jan 20 19:45:16.777 INF net Adding pubsub topic PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.778 INF net Publish log PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.780 INF net Adding pubsub topic PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.781 INF net Publish log PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.782 INF net Adding pubsub topic PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW Topic=bafkreifbk3dtij7vgjhm7xow5i2hnhw5ppieityb2eklzwdst3yph7h4p4
Jan 20 19:45:16.782 INF net Received new pubsub event PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW SenderId=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512 Message=JOINED
Jan 20 19:45:16.783 INF net Publish log PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW Topic=bafkreifbk3dtij7vgjhm7xow5i2hnhw5ppieityb2eklzwdst3yph7h4p4
Jan 20 19:45:16.784 INF net Received pushlog PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW Creator=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW DocID=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.786 INF net Adding pubsub topic PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bafkreifbk3dtij7vgjhm7xow5i2hnhw5ppieityb2eklzwdst3yph7h4p4
Jan 20 19:45:16.788 INF net Received new pubsub event PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW SenderId=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.788 INF net Received pushlog PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Creator=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea DocID=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.788 INF net DAG sync complete PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea DocID=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.789 INF net Adding pubsub topic PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.789 INF net DAG sync complete PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW DocID=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.790 INF net Publish log PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bafkreifbk3dtij7vgjhm7xow5i2hnhw5ppieityb2eklzwdst3yph7h4p4
Jan 20 19:45:16.791 INF net Adding pubsub topic PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512
Jan 20 19:45:16.795 INF db Closing DefraDB process...
Jan 20 19:45:16.796 INF db Successfully closed running process
Jan 20 19:45:16.796 INF net Removing all pubsub topics PeerID=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea
Jan 20 19:45:16.797 INF net Received new pubsub event PeerID=12D3KooWLxzXgoDC1TwWVCpb27gF6x4WVDriNF4R9GXGmZ7uCLrW SenderId=12D3KooWA4XDFcRgLa5V5xnYXi1j6NowxBgPPky2qaBNAQ3mxJea Topic=bae-848eab20-3ade-55d9-b16a-99a96ae89512 Message=LEFT
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x5a8060e]

goroutine 8065 [running]:
github.com/sourcenetwork/defradb/internal/db/fetcher.(*wrappingFetcher).FetchNext(0xc003106d80, {0x9181270, 0xc0052d03f0})
	/home/runner/work/defradb/defradb/internal/db/fetcher/wrapper.go:151 +0xae
github.com/sourcenetwork/defradb/internal/lens.(*lensedFetcher).FetchNext(0xc004f99340, {0x9181270, 0xc0052d03f0})
	/home/runner/work/defradb/defradb/internal/lens/fetcher.go:133 +0xc4
github.com/sourcenetwork/defradb/internal/db.(*collection).get(0xc004f8c480, {0x9181270, 0xc0052d03f0}, {0x387acf0?, {0xc00387acf0?, 0x4f0e89?}}, {0x0, 0x0, 0x0}, 0x0)
	/home/runner/work/defradb/defradb/internal/db/collection_get.go:82 +0x64b
github.com/sourcenetwork/defradb/internal/db.(*collection).Get(0xc004f8c480, {0x9181270, 0xc0052d03c0}, {0x1, {0x84, 0x8e, 0xab, 0x20, 0x3a, 0xde, ...}, ...}, ...)
	/home/runner/work/defradb/defradb/internal/db/collection_get.go:44 +0x478
github.com/sourcenetwork/defradb/internal/db.syncIndexedDoc({0x9181270, 0xc00576d770}, {0x1, {0x84, 0x8e, 0xab, 0x20, 0x3a, 0xde, 0x55, ...}, ...}, ...)
	/home/runner/work/defradb/defradb/internal/db/merge.go:572 +0x14a
github.com/sourcenetwork/defradb/internal/db.(*DB).executeMerge(0xc0018b1440, {0x91812a8, 0xc002770410}, 0xc004f8c480, {{0xc00374e0f0, 0x28}, {0xc00374e1e0, 0x26}, {0xc00374e150, 0x26}, ...})
	/home/runner/work/defradb/defradb/internal/db/merge.go:81 +0x9a6
github.com/sourcenetwork/defradb/internal/db.(*DB).handleMessages.func1()
	/home/runner/work/defradb/defradb/internal/db/messages.go:69 +0x7fe
created by github.com/sourcenetwork/defradb/internal/db.(*DB).handleMessages in goroutine 7556
	/home/runner/work/defradb/defradb/internal/db/messages.go:41 +0xad6
```

--- a/internal/db/fetcher/wrapper.go
+++ b/internal/db/fetcher/wrapper.go
@@ -123,13 +123,13 @@ func (f *wrappingFetcher) Start(ctx context.Context, prefixes ...keys.Walkable) 
 	var top fetcher
 	top, err = newPrefixFetcher(ctx, f.txn, dsPrefixes, f.col, fieldsByID, client.Active, &execInfo)
 	if err != nil {
-		return nil
+		return err
 	}
 
 	if f.showDeleted {
 		deletedFetcher, err := newPrefixFetcher(ctx, f.txn, dsPrefixes, f.col, fieldsByID, client.Deleted, &execInfo)
 		if err != nil {
-			return nil
+			return err
 		}
 
 		top = newMultiFetcher(top, deletedFetcher)

